src/Workspaces/Core/Portable/CodeFixes/CodeFixProvider.cs

```diff
@@ -38,20 +41,30 @@ public abstract class CodeFixProvider
             => null;
 
         /// <summary>
-        /// What priority this provider should run at.
+        /// Computes the <see cref="CodeActionRequestPriority"/> group this provider should be considered to run at. Legal values
+        /// this can be must be between <see cref="CodeActionRequestPriority.Low"/> and <see cref="CodeActionPriority.High"/>.
         /// </summary>
-        internal CodeActionRequestPriority RequestPriority
+        /// <remarks>
+        /// Values outside of this range will be clamped to be within that range.  Requests for <see
+        /// cref="CodeActionRequestPriority.High"/> may be downgraded to <see cref="CodeActionRequestPriority.Default"/> as they
+        /// poorly behaving high-priority providers can cause a negative user experience.
+        /// </remarks>
+        protected virtual CodeActionRequestPriority ComputeRequestPriority()
+            => CodeActionRequestPriority.Default;
+
+        /// <summary>
+        /// Priority class this refactoring provider should run at. Returns <see
+        /// cref="CodeActionRequestPriority.Default"/> if not overridden.  Slower, or less relevant, providers should
+        /// override this and return a lower value to not interfere with computation of normal priority providers.
+        /// </summary>
+        public CodeActionRequestPriority RequestPriority
         {
             get
             {
                 var priority = ComputeRequestPriority();
-                // Note: CodeActionRequestPriority.Lowest is reserved for IConfigurationFixProvider.
-                Contract.ThrowIfFalse(priority is CodeActionRequestPriority.Low or CodeActionRequestPriority.Normal or CodeActionRequestPriority.High);
-                return priority;
+                Debug.Assert(priority is CodeActionRequestPriority.Low or CodeActionRequestPriority.Default or CodeActionRequestPriority.High, "Provider returned invalid priority");
```

Comment:
note: this is just an assert to catch if we do something wrong ourselves.  We dont' want to throw here as our approach is to just clamp.  that way someone coudl request a higher pri than currently supported, and we could potentially allow it in the future.